Library/Homebrew/software_spec.rb

```diff
@@ -47,6 +47,7 @@ def initialize
   def owner= owner
     @name = owner.name
     @full_name = owner.full_name
+    @dependency_collector.tap = owner.tap
```

Comment:
You need to be aware that the tap here is not `Tap` class but result of  `Formula#tap`. It can return following results:
* `nil`: unknown origin we should treat it as core.
* `"Homebrew/homebrew"`: core.
* `"<user>/homebrew-<repo>"`: tap. You probably want to remove the `homebrew-`

It can be a little messy here. If you want, you can try to clean them.(e.g. turn them to `Tap` object)